lib/Scheduler/Scheduler.cpp

```diff
@@ -0,0 +1,97 @@
+// SPDX-FileCopyrightText: (c) 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "ttmlir/Scheduler/Scheduler.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "ttmlir/Dialect/TTIR/IR/TTIROpsDialect.h.inc"
+#include <mlir/IR/BuiltinOps.h>
+#include <mlir/IR/Operation.h>
+#include <mlir/Interfaces/DestinationStyleOpInterface.h>
+
+namespace mlir::tt::scheduler {
+
+bool isTTIROp(mlir::Operation *op) {
+  return isa<ttir::TTIRDialect>(op->getDialect());
+}
+
+bool isDpsOp(mlir::Operation *op) {
+  return isa<mlir::DestinationStyleOpInterface>(op);
+}
+
+// Init the dependencies map of all TTIR ops
+// which implement DestinationStyleOpInterface
+// We check for DPS because all operations which do
+// some computation in TTIR implement this interface
+Scheduler::Scheduler(func::FuncOp *func) {
+  for (auto &op : func->getOps()) {
+    if (isTTIROp(&op) && isDpsOp(&op)) {
+      dependencies[&op] = {};
```

Comment:
Is it possible to use op->getOperands() instead of having dependencies as part of scheduler state?